public/New-DbaDbUser.ps1

```diff
@@ -88,136 +91,127 @@ function New-DbaDbUser {
+            $UserName = $Login
+        }
+
+        #Set appropriate user type
+        #Removed SQLLogin  user type. This is deprecated and the alternate is to map to SqlUser. Reference https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.smo.usertype?view=sql-smo-160
+        if ($ExternalProvider) {
+            Write-Message -Level Verbose -Message "Using UserType: External"
+            $userType = [Microsoft.SqlServer.Management.Smo.UserType]::External
+        } elseif ($Password -or $Login) {
+            Write-Message -Level Verbose -Message "Using UserType: SqlUser"
+            $userType = [Microsoft.SqlServer.Management.Smo.UserType]::SqlUser
+        } else {
+            Write-Message -Level Verbose -Message "Using UserType: NoLogin"
+            $userType = [Microsoft.SqlServer.Management.Smo.UserType]::NoLogin
         }
     }
 
     process {
-        if (-not $Login -and -not $Username) {
-            Stop-Function -Message "One of -Login or -Username is needed."
-            return
-        }
 
         foreach ($instance in $SqlInstance) {
-            try {
-                $server = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
-            } catch {
-                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            #prepare parameter values
+            $connParam.SqlInstance = $instance
+            $getDbParam = $connParam.Clone()
+            $getDbParam.OnlyAccessible = $True
+            if ($Database) { $getDbParam.Database = $Database }
+            if ($ExcludeDatabase) { $getDbParam.ExcludeDatabase = $ExcludeDatabase }
+            if (-not ($IncludeSystem)) { $getDbParam.ExcludeSystem = $True }
+
+            # Is the login exist?
+            if ($Login -and (-not(Get-DbaLogin @connParam -Login $Login))) {
+                Stop-Function -Message "Invalid Login: $Login is not found on $instance, skipping." -Target $instance -Continue
             }
 
-            # Does the given login exist?
-            if ($Login) {
-                $existingLogin = $server.Logins | Where-Object Name -eq $Login
-                if (-not $existingLogin) {
-                    Stop-Function -Message "Invalid Login: $Login is not found on $instance, skipping." -Target $instance -Continue
-                }
-            }
-
-            $databases = $server.Databases | Where-Object IsAccessible -eq $true
-
-            if ($Database) {
-                $databases = $databases | Where-Object Name -In $Database
-            } else {
-                if (-not $IncludeSystem) {
-                    $databases = $databases | Where-Object IsSystemObject -ne $true
-                }
-            }
-            if ($ExcludeDatabase) {
-                $databases = $databases | Where-Object Name -NotIn $ExcludeDatabase
-            }
+            $databases = Get-DbaDatabase @getDbParam
```

Comment:
We tend to not use this in our commands internally because of performance impact. I would prefer this to be reverted to the original pattern as that is what is followed in most all of our commands.

Reply:
Sorry I didnt get it. You meant the hash table as parameters or using the `Get-DbaDatabase` function?

Reply:
My request is reverting the chnage back to the original pattern. `Get-DbaDatabase` incurs performance on many systems and use of that in any command can cause degradation.

Reply:
Gotcha. Before making this change I raised in the Slack channel and I was told there isn't any standard and suggested to use functions if there is one already exists. That's when I refactored to use the functions instead.

I'm happy to revert the changes to use SMO connection instead of the function. Since this PR is merged, do you guys want me to revert it and raise a new PR?

Also, this is my understanding. These are operational functions and not a real time application so it should be fine even if it add few ms to the execution. I know this module is used in wide scope so I dont want to mess up with the expectation. Like I said before, let me know if I need to revert it and create a new PR?